docs/source/contributing.rst

```diff
@@ -4,88 +4,139 @@ Contributing
+   click the "fork" button to create your own copy of the project. You will
+   need a GitHub account. This will allow you to make a "pull request" (PR)
    to the ManimCommunity repo later on.
 
-#. *Clone your fork.*
-   You can clone your Github fork by running:
+#. Clone your fork to your local computer:
 
    .. code-block:: shell
 
-      git clone <your-fork-url>
-      cd manim
+      git clone https://github.com/<your-username>/manim.git
 
    GitHub will provide both a SSH (``git@github.com:<your-username>/manim.git``) and
-   HTTPS (``https://github.com/<your-username>/manim.git``) URL for cloning the repo.
-   You can use whichever one you are setup for.
+   HTTPS (``https://github.com/<your-username>/manim.git``) URL for cloning.
+   You can use SSH if you have SSH keys setup.
 
    .. WARNING::
 
-      Do not ``git clone`` the original ManimCommunity repository.  You must
-      clone your own fork.  After this step, there are three different
-      repositories to keep track of: the original ManimCommunity repo, your own
-      fork of it, and your local repository.
+      Do not clone the ManimCommunity repository. You must clone your own
+      fork.
+
+#.  Change the directory to enter the project folder:
+
+    .. code-block:: shell
+
+       cd manim
 
-#. Make ``git`` aware of the ManimCommunity repo.
+#. Add the upstream repository, ManimCommunity:
 
    .. code-block:: shell
 
       git remote add upstream https://github.com/ManimCommunity/manim.git
-      git fetch upstream
+ 
+#. Now, ``git remove -v`` should show two remote repositores named:
+   
+   - ``origin``, your forked repository
+   - ``upstream`` the ManimCommunity repository
 
-   After these commands, your local repository can keep track of your fork
-   (referred to as 'origin') as well as the main ManimCommunity repository
-   (referred to as 'upstream').
+#. Install Manim:
+ 
+   - See our :doc:`instructions for developers<installation/for_dev>` for
+     details and continue here afterwards
 
-#. *Install manim.*
-   See the :doc:`installation instructions for developers <installation/for_dev>` for
-   details.
+#. Install Pre-Commit (recommended):
 
-You are now ready to work on manim!
+   .. code-block:: shell
+
+      pre-commit install
+   
+   This is optional and will ensure during development that each of your
+   commit is properly formatted against our formatter, ``black``.
 
+You are now ready to work on manim!
 
-Changing manim's source code
-============================
+Develop your contribution
+--------------------------
 
-#. *Choose the branch for your changes.*
-   To work on the ManimCommunity master branch, you can change to it with:
+#. Checkout your local repository's master branch and pull the latest
+   changes from ManimCommunity, `upstream`, into your local repository:
 
    .. code-block:: shell
 
-      git checkout -b master upstream/master
+      git checkout master 
+      git pull upstream master
```

Comment:
this will sadly create a merge commit :(

Reply:
we should recommend people to check out a branch from `upstream/master`

Reply:
I just followed NumPy documentation, but also I've faced issues in the past with editing people's PRs because their PR is made on `master` which is protected. Instead it should be better to make the user use a descriptive name not on master/main/ a protected branch

Reply:
As for the merge commit, I'm okay with that since they all get squashed anyways, and the user can always `git commit --amend` or `git rebase -i HEAD~1`

Reply:
or they can do 
```suggestion
      git pull --rebase upstream master
```

Reply:
Sure, I'm fine with this